legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/protocols/pure/vX_X_X/invocations/execution_relational_testConnection.pure

```diff
@@ -151,7 +153,19 @@ function meta::protocols::pure::vX_X_X::transformation::toPureGraph::connection:
             // _type = 'gcpWorkloadIdentityFederationWithAWS',
             serviceAccountEmail = $b.serviceAccountEmail,
             additionalGcpScopes = $b.additionalGcpScopes
-         )
+         ),
+      a:meta::protocols::pure::vX_X_X::metamodel::store::relational::connection::alloy::authentication::AwsPKAuthenticationStrategy[1] |
+        ^meta::pure::alloy::connections::alloy::authentication::AwsPKAuthenticationStrategy(
```

Comment:
This pk model is different from the existing pk model for Snowflake ...       s:meta::pure::alloy::connections::alloy::authentication::SnowflakePublicAuthenticationStrategy[1] |

```       ^meta::protocols::pure::vX_X_X::metamodel::store::relational::connection::alloy::authentication::SnowflakePublicAuthenticationStrategy(
            _type = 'snowflakePublic',
            privateKeyVaultReference = $s.privateKeyVaultReference,
            passPhraseVaultReference = $s.passPhraseVaultReference,
            publicUserName = $s.publicUserName
        ),
```

Apart from the "public" in the name, the main difference is that the Snowflake model is explicit about using an encrypted private key (hence the pass phrase attribute). 

In your model are you abstracting that behind the secret ARN/AWS ?

Reply:
This is a good point .. goes to the discussion we were having yesterday ... we could either implicitly interpret what's in there as an ARN or create a new scheme that explicitly declares that its an ARN. Or further categorize different forms of Vault references. In the SnowflakePublicAuthenticationStrategy scheme, both private key and pass phrase are required. Should we make the latter optional? I am okay with removing AwsPKAuthenticationStrategy, making passPhraseVaultReference be [0..1] in SnowflakePublicAuthenticationStrategy and - for now - implicitly expecting an ARN there when used with cloud persistence